akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -220,4 +220,11 @@ object Sink extends SinkApply {
     new Sink(new AcknowledgeSink(bufferSize, DefaultAttributes.acknowledgeSink, shape("AcknowledgeSink"), timeout))
   }
 
+  /**
+   * Crates a `Sink` that is materialized into `Source` to connect with another flow.
+   * This sink gets backpressure from second flow and does not have internal buffer.
+   */
+  def pipedSink[T](): Sink[T, Source[T, Unit]] =
+    Sink.publisher[T].mapMaterializedValue(Source(_)).named("PipedSink")
```

Comment:
Guys, I think you might be partially talking past each other. Perhaps it helps to add a completely different viewpoint: on the topic of whether to accept a Source or provide a Sink there is no universally preferred answer. Whoever provides the API needs to make a choice about which party shall be in control of the materialization (i.e. the setting in motion of things). For Akka HTTP this decision was pretty simple in that it is much more difficult to do the wrong thing when the API demands a Source—for one it is impossible to forget to materialize a provided Sink (disregarding the other technical difficulties that this alternative would pose).

In a sense this decision is very similar to toolkit vs. framework: a toolkit will provide tools to be assembled by the user, a framework will require the user to provide the customizable pieces and build the application from those. In this sense Akka HTTP is more framework than toolkit—to a degree.

In any case I think this discussion has long since strayed from its context (the PR) and I do not see any reason to change my conclusion on the code itself.